File: src_fortran/sub/maxri.f
c
c
c
c
c
c
      subroutine MAXRI(n,x,xmax,jndex)
c determine maximum-value of a real-array
      implicit none
      integer i,n,jndex
      real x(n), xmax
      jndex=1
      do 1 i=1,n
   1  if(x(jndex).le.x(i)) jndex=i
      xmax=x(jndex)
      return
      end ! of subr. maxri

File: src/sub/maxri.c
int maxri(int n, const double *x, double *xmax, int *jndex) {
    int i;

    if (n <= 0 || x == 0 || xmax == 0 || jndex == 0) {
        return -1;
    }

    *jndex = 0;
    for (i = 0; i < n; ++i) {
        if (x[*jndex] <= x[i]) {
            *jndex = i;
        }
    }

    *xmax = x[*jndex];
    return 0;
}
